### PR TITLE
New updates of format and indentation of Extra algorithm, included test ...

### PR DESCRIPTION
**Pull request by lfloresayaviri**
...cases and updates on the test suite
Hello Edson,
code updated according to your imputs,
Please, let me know your comments
Thanks

### CODE REVIEW

File: src/solver/algorithms/Fallin_solver.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+            for j in range(9):
+                self.compare_cell(i,j,data)
+                
+    def dump(self):
+        '''Dumps to external data (ie, a simple matrix/array-of-arrays)'''
+        out = []
+        for i in range(9):
+            out += [ [] ]
+            for j in range(9):
+                self.out_cell(i,j,out)
```
Comment (edson-gonzales) [inline src/solver/algorithms/Fallin_solver.py:41]: spaces after commas

File: src/solver/algorithms/Fallin_solver.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+        copy = FallinSolver()
+        for i in range(9):
+            for j in range(9):
+                copy.cells[i][j] = self.cells[i][j]
+                copy.count[i][j] = self.count[i][j]
+                copy.known = self.known
+        return copy
+
+    def load(self, data):
+        '''Loads from external data'''
```
Comment (edson-gonzales) [inline src/solver/algorithms/Fallin_solver.py:29]: there is no documentation for the data parameter

File: src/solver/algorithms/Fallin_solver.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    def dump(self):
+        '''Dumps to external data (ie, a simple matrix/array-of-arrays)'''
+        out = []
+        for i in range(9):
+            out += [ [] ]
+            for j in range(9):
+                self.out_cell(i,j,out)        
+        return out
+        
+    def load_string(self, str):
```
Comment (edson-gonzales) [inline src/solver/algorithms/Fallin_solver.py:44]: there is no documentation for the str parameter

File: src/solver/algorithms/Fallin_solver.py [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+        for i in range(9):
+            self.cells += [ [] ]
+            self.count += [ [] ]
+            for j in range(9):
+                self.cells[i] += [0]
+                self.count[i] += [0]
+        self.known = 0
+    
+    def copy(self):
+        '''Makes a copy of the sudoku data'''
```
Comment (edson-gonzales) [inline src/solver/algorithms/Fallin_solver.py:19]: No documentation for the return parameter

File: src/solver/algorithms/Fallin_solver.py [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+            line = line.strip()            
+            if(line == ''):
+                continue
+            row = []            
+            block_of_three = line.split()
+            self.verify_block_of_three(block_of_three,line,row)          
+            lines += [row]
+        self.load(lines)
+    
+    def verify_block_of_three(self,block_of_three,line,row):
```
Comment (edson-gonzales) [inline src/solver/algorithms/Fallin_solver.py:62]: spaces after commas

File: src/solver/algorithms/Fallin_solver.py [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+        '''Verify in a block of three'''
+        for field in block_of_three:
+            field = field.strip()
+            if(field == ''):
+                continue
+            element = int(field)
+            row += [element]
+
+    def dump_string(self):
```
Comment (edson-gonzales) [inline src/solver/algorithms/Fallin_solver.py:71]: why we have methods like dump and dump_string, load and load_string?

File: src/solver/algorithms/Fallin_solver.py [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+        '''Release the solution on an string format''' 
+        out = self.dump()
+        out_str = ""
+        for i in range(9):
+            for j in range(9):
+                out_str += str(out[i][j]) + " "
+                out_str=self.concat_spaces(j,out_str)        
+            out_str += "\n"
+            if(i % 3 == 2): out_str += "\n"
```
Comment (edson-gonzales) [inline src/solver/algorithms/Fallin_solver.py:80]: do not use inline notation

File: src/solver/algorithms/Fallin_solver.py [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+        return out_str
+        
+    def concat_spaces(self,j,out_str):
+        '''Concatenate spaces'''
+        if(j % 3 == 2): out_str += "  "
```
Comment (edson-gonzales) [inline src/solver/algorithms/Fallin_solver.py:85]: Do not use inline notation

File: src/solver/algorithms/Fallin_solver.py [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+    def reduce(self):
+        '''finds possible elements for each position'''
+        self.known = 0        
+        for i in range(9):
+            self.check_has_solution(i) 
+        ''' done: successful (ie, no impossible entries)'''
+        return 1
+
+    def check_has_solution(self,i):
+        for j in range(9):
```
Comment (edson-gonzales) [inline src/solver/algorithms/Fallin_solver.py:102]: no documentation

**Comment by edson-gonzales** [comment]
Please review the documentation of all the methods, especially parameters should have examples and the same for the return values of the methods, after that we will need another review

**Comment by lfloresayaviri** [comment]
Hi Edson, 
I updated the code according to your inputs,
Please, let me know your comments

**Pull request closed without merging**